samples/bluetooth/bap_broadcast_sink/src/main.c

```diff
@@ -91,12 +91,17 @@ static const struct bt_audio_codec_cap codec_cap = BT_AUDIO_CODEC_CAP_LC3(
 	CONFIG_MAX_CODEC_FRAMES_PER_SDU,
 	(BT_AUDIO_CONTEXT_TYPE_CONVERSATIONAL | BT_AUDIO_CONTEXT_TYPE_MEDIA));
 
-/* Create a mask for the maximum BIS we can sync to using the number of streams
- * we have. Bit 0 is BIS index 1.
- */
-static const uint32_t bis_index_mask = BIT_MASK(CONFIG_BT_BAP_BROADCAST_SNK_STREAM_COUNT);
+struct bis_index_data {
+	uint32_t bis_index_bitfield[CONFIG_BT_BAP_BASS_MAX_SUBGROUPS];
+	enum bt_audio_location audio_allocation[BT_ISO_BIS_INDEX_MAX];
+	uint8_t num_subgroups;
+};
+
 static uint32_t requested_bis_sync;
-static uint32_t bis_index_bitfield;
+/* Bit field indicating from which subgroup(s) BIS sync is requested */
+static uint32_t requested_subgroup_sync;
+static uint32_t base_bis_index_bitfield;
+static struct bis_index_data bis_index;
```

Comment:
We should probably also consider that there are risks of race conditions with this increased use of global variables that are accessed by multiple threads